fix(modal): Make modal keyboard accessible

LZoog opened the pull request:
Fixes #597, #636, #2869

@shane-tomlinson I'd like you to take a look at this when you can since you'd reviewed #724, I have a question around testing... Should this be taken out of unit tests and placed in avatar & client disconnection selenium tests? Or how would you test these changes?

Also, I created #2869 while working on these issues, but I'm actually having trouble getting into the modal with a screenreader cursor to benefit from the context created there with `aria-labelledby`, but I asked UX if they'd like me to leave that header in there anyway and they confirmed it should be left. I'll take a better look at this accessibility-wise when/after we do the settings redesign. Ideally, we use `aria-hidden` and other `aria` properties, but this issue fix is to make the modal _keyboard_ accessible and not necessarily screenreader accessible, which will be on my radar for settings redesign.

<img width="400" alt="image" src="https://user-images.githubusercontent.com/13018240/66847541-28e85580-ef39-11e9-81a4-02bb34e6a7e3.png">
Text spacing was given the r+ from Julia and Janice. Screenshot shows what "focus" looks like when tabbing around (which is the same as hovering).

shane-tomlinson commented on packages/fxa-content-server/app/scripts/views/mixins/modal-panel-mixin.js:
```diff
@@ -33,8 +36,66 @@ export default {
+      event.keyCode === KeyCodes.ENTER &&
+      document.activeElement.getAttribute('type') === 'radio'
+    ) {
+      event.preventDefault();
```
The behavior here seems slightly strange to me when testing it, I suppose I assumed tab would select the radio button. How hard would it be to submit the form if the radio button is already selected, and select if not?

shane-tomlinson commented on packages/fxa-content-server/app/scripts/templates/settings/client_disconnect.mustache:
```diff
@@ -1,14 +1,16 @@
 
       <form novalidate>
+        <div role="radiogroup" aria-labelledby="radio-header">
+          <h3 id="radio-header" class="radio-header">{{#t}}You are disconnecting %(deviceName)s from Sync because this device is:{{/t}}</h3>
```
The wording "this device is" makes the radio button labels seem off, .e.g.:

"this device is:" "Suspicious device"

The "device" on the end of "suspicious" seems somewhat superfluous.

LZoog replied:
@janiceahn @janiceahn Agreed, I didn't notice this. Would you like to tweak this copy?

shane-tomlinson commented on packages/fxa-content-server/app/scripts/views/mixins/modal-panel-mixin.js:
```diff
@@ -33,8 +36,66 @@ export default {
+
+    //avoid IME composition keydown events
+    //ref: https://developer.mozilla.org/docs/Web/Events/keydown#Notes
+    if (event.isComposing || event.keyCode === 229) return;
```
Can you add `229` to [key-codes.js](https://github.com/mozilla/fxa/blob/0bbac2ba3666749309d9bfc03448fc38a1b21331/packages/fxa-content-server/app/scripts/lib/key-codes.js#L4) and use a named constant here?

Also, can you put the `return` on its own line and add some block`{}`?

I thought we had a lint rule about that.

LZoog replied:
Done. Haha, developer preference. I don't even use semicolons on personal projects -le gasp-

shane-tomlinson commented:
Some more thoughts, I think @ryanfeeley wants to get rid of the modal dialogs all together because of the UI issues around them, I wonder if its worthwhile holding off to find out how much effort this warrants? If removing the modals is more than a few months away, getting this merged with a couple of basic tests is probably still worthwhile.

LZoog commented:
@shane-tomlinson I addressed all of your comments besides the one I'll need UX input on for copy phrasing. Your camera close issue stemmed from me changing the `a[href]` selector to `#file` and `#camera` when I was working with that test template but I think these are going to have to be seleinum tests so I can revert those and leave the selector as `a[href]`.

The form now submits when you hit "enter" on a radio button already selected - this is a good suggestion IMO, just from using screenreaders and such before I personally wouldn't expect a tab selection to select for you because users should be able to tab to a submit button if they want to. It's a can of worms like you said though, someone else could argue for some other functionality. 🤷‍♀️ Regardless I think what you suggested is a good option.

I'm going to have to put getting tests in on hold until I get through a couple of train 149 priorities first. * old man yells at cloud emoji *

LZoog commented:
@shane-tomlinson 

Since it doesn't look like [the simulated events (here Tab) does not trigger browser functionality](https://github.com/theintern/leadfoot/issues/98#issuecomment-264970531), specifically advance focus to next element in tab order, this doesn't seem to work as a Selenium test, and making `test_template.mustache` pop open a modal to test seems overly complicated since we are redesigning settings anyway. I left a single test to check for the active element after the tab is open... I know this doesn't have super great code coverage, but I think this should be fine as-is, considering the upcoming redesign. 🤷‍♀️
